test.disabled/ejabberd_tests/tests/dynamic_modules.erl

```diff
@@ -10,25 +10,29 @@ save_modules(Domain, Config) ->
 
 ensure_modules(Domain, RequiredModules) ->
     CurrentModules = get_current_modules(Domain),
-    [ensure_module(Domain, Mod, Opts, proplists:get_value(Mod, CurrentModules, stopped))
-        || {Mod, Opts} <- RequiredModules].
+    {ToReplace, ReplaceWith} = to_replace(RequiredModules, CurrentModules, [], []),
+    ok = escalus_ejabberd:rpc(gen_mod_deps, replace_modules,
+                              [Domain, ToReplace, ReplaceWith]).
+
+to_replace([], _CurrentModules, ReplaceAcc, ReplaceWithAcc) ->
+    {lists:usort(ReplaceAcc), ReplaceWithAcc};
+to_replace([RequiredModule | Rest], CurrentModules, ReplaceAcc, ReplaceWithAcc) ->
+    {Mod, Opts} = RequiredModule,
+    {NewReplaceAcc, NewReplaceWithAcc} =
+        case lists:keyfind(Mod, 1, CurrentModules) of
+            false when Opts =:= stopped -> {ReplaceAcc, ReplaceWithAcc};
+            false -> {ReplaceAcc, [RequiredModule | ReplaceWithAcc]};
+            {Mod, Opts} -> {ReplaceAcc, ReplaceWithAcc};
+            ExistingMod when Opts =:= stopped -> {[ExistingMod | ReplaceAcc], ReplaceWithAcc};
```

Comment:
Will this ever be executed?

Reply:
I'm not sure if you're asking if anyone will ever call `ensure_modules` with `stopped` opts for a module? Already happens in some commercial projects, I'm sure it will come up.

Reply:
OK